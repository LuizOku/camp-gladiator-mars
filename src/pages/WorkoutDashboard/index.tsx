import React, { useEffect } from "react";
import { getWorkouts } from "../../api/workout";

const WorkoutDashboard: React.FC = () => {
  useEffect(() => {
    const fetchWorkouts = async () => {
      const res = await getWorkouts();
      console.log(res);
    };
    fetchWorkouts();
  }, []);
  return <div>WorkoutDashboard</div>;
};

export default WorkoutDashboard;
