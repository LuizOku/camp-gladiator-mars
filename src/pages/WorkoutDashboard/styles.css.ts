import styled from "styled-components";
import { Heading } from "@campgladiator/cgui-core.atoms.typography";
import { Button } from "@campgladiator/cgui-core.atoms.button";
import { Fieldset } from "@campgladiator/cgui-core.atoms.fieldset";
import { Select } from "@campgladiator/cgui-core.molecules.select";

export const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #f9f9f9;
  padding: 100px;
`;

export const Text = styled(Heading)`
  color: #ffffff;
  margin-top: 10px;
`;

export const StyledSelect = styled(Select)`
  margin: 10px;
  width: 200px;
`;

export const StyledButton = styled(Button)`
  width: 200px;
  margin: 10px;
`;

export const StyledFieldset = styled(Fieldset)`
  margin-top: 50px;
  margin-bottom: 50px;
`;

export const CardsContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
`;
