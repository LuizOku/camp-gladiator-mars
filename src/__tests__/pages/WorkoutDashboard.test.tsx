import { render, screen, act } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";

import WorkoutDashboard from "../../pages/WorkoutDashboard";
import { setupStore } from "../../store";
import "../../setupTests";

const workouts = [
  {
    description:
      "Enjoy a fun and high energy workout with Amber. We will engage a full body workout and get your heart rate up!",
    thumbnail:
      "https://cgcdn.s3.amazonaws.com/nation/users/UID-W1405610919UID-53c7eba7a992b.png",
    levelTag: "beginner",
    media: "https://virtual-library.s3.amazonaws.com/Week2-AmberAutrey.mp4",
    title: "Strength and Agility by Amber Autrey",
    impactTag: "high",
    id: "3",
    trainerId: "3",
    duration: 60,
  },
  {
    description:
      "Enjoy a fun and low impact workout with Claire. We will engage a full body workout and get your heart rate up!",
    thumbnail:
      "https://cgcdn.s3.amazonaws.com/nation/users/UID-D1489527062UID-58c86116311d9.png",
    levelTag: "beginner",
    media: "https://virtual-library.s3.amazonaws.com/Week2-ClaireDonahoe.mp4",
    title: "Strength and Agility by Claire Donahoe",
    impactTag: "low",
    id: "4",
    trainerId: "4",
    duration: 60,
  },
];

const promise = Promise.resolve();

const renderComponent = () => {
  const store = setupStore({
    workout: {
      list: workouts,
    },
  });
  return render(
    <Provider store={store}>
      <Router>
        <WorkoutDashboard />
      </Router>
    </Provider>
  );
};

describe("<WorkoutDashboard />", () => {
  it("renders without crashing", async () => {
    const view = renderComponent();
    expect(view).toMatchSnapshot();
    await act(async () => {
      await promise;
    });
  });

  it("should have the hero section", async () => {
    renderComponent();
    expect(screen.getByTestId("hero-section")).toBeInTheDocument();
    await act(async () => {
      await promise;
    });
  });

  it("should have the title", async () => {
    renderComponent();
    expect(
      screen.getByText("Find the exercise that fits to you")
    ).toBeInTheDocument();
    await act(async () => {
      await promise;
    });
  });

  it("should have the level select of the filter", async () => {
    renderComponent();
    expect(screen.getByText("Level")).toBeInTheDocument();
    await act(async () => {
      await promise;
    });
  });

  it("should have the impact select of the filter", async () => {
    renderComponent();
    expect(screen.getByText("Impact")).toBeInTheDocument();
    await act(async () => {
      await promise;
    });
  });

  it("should have the Filter button", async () => {
    renderComponent();
    expect(screen.getByText("Filter")).toBeInTheDocument();
    await act(async () => {
      await promise;
    });
  });

  it("should have the workout cards", async () => {
    renderComponent();
    for (const work of workouts) {
      expect(screen.getByTestId(`workout-card-${work.id}`)).toBeInTheDocument();
      await act(async () => {
        await promise;
      });
    }
  });
});
