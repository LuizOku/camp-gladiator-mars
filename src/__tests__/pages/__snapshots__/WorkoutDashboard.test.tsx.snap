// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkoutDashboard /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dkzDqf uWKzM"
        data-testid="hero-section"
      >
        <h3
          class="heading sc-hKMtZM eZJtHV"
          data-font="united"
          data-variation="display"
        >
          Camp Gladiator is going to Mars...
        </h3>
        <h3
          class="heading sc-hKMtZM eZJtHV"
          data-font="united"
          data-variation="default"
        >
          Start now your virtual workouts!
        </h3>
      </div>
      <div
        class="sc-papXJ czGCkr"
      >
        <h3
          class="heading "
          data-font="gotham"
          data-variation="display"
        >
          Find the exercise that fits to you
        </h3>
        <form>
          <div
            class="fieldset fieldset--inline sc-crXcEl iRbwfs"
          >
            <div
              class="content sc-kDDrLX jUNfll"
            >
              <div
                class="outsideWrapper active"
                data-disabled="false"
                data-invalid="false"
                style="max-height: 75px;"
              >
                <div
                  class="innerWrapper noRecords"
                >
                  <p
                    class="paragraph placeholder"
                    data-size="small"
                    data-weight="book"
                  >
                    Level
                  </p>
                  <div
                    class="wrapper paragraph"
                    data-size="small"
                    data-value=""
                  >
                    <input
                      class="input"
                      value=""
                    />
                  </div>
                  <div
                    class="controls"
                  />
                </div>
                <input
                  hidden=""
                  name="level"
                  readonly=""
                  value=""
                />
              </div>
            </div>
            <div
              class="content sc-kDDrLX jUNfll"
            >
              <div
                class="outsideWrapper active"
                data-disabled="false"
                data-invalid="false"
                style="max-height: 75px;"
              >
                <div
                  class="innerWrapper noRecords"
                >
                  <p
                    class="paragraph placeholder"
                    data-size="small"
                    data-weight="book"
                  >
                    Impact
                  </p>
                  <div
                    class="wrapper paragraph"
                    data-size="small"
                    data-value=""
                  >
                    <input
                      class="input"
                      value=""
                    />
                  </div>
                  <div
                    class="controls"
                  />
                </div>
                <input
                  hidden=""
                  name="impact"
                  readonly=""
                  value=""
                />
              </div>
            </div>
            <button
              class="button sc-iqcoie fwqNIQ"
              data-children="true"
              data-emphasis="primary"
              data-floating="false"
              data-icon="false"
              data-size="default"
              data-variation="solid"
              type="submit"
            >
              <span
                class="label content"
                data-size="default"
              >
                Filter
              </span>
            </button>
          </div>
        </form>
        <div
          class="sc-evZas gzZmBv"
        >
          <article
            class="card false card--raised sc-eCYdqJ fmqJBb"
            data-testid="workout-card-3"
          >
            <img
              class="avatar sc-gKXOVf efHPNN"
              data-testid="thumbnail-Strength and Agility by Amber Autrey"
              src="https://cgcdn.s3.amazonaws.com/nation/users/UID-W1405610919UID-53c7eba7a992b.png"
              style="width: 50px; height: 50px;"
            />
            <div
              class="sc-jSMfEi hEqNgO"
            >
              <p
                class="paragraph "
                data-size="default"
                data-weight="bold"
              >
                Strength and Agility by Amber Autrey
              </p>
              <div
                class="sc-iBkjds cdystf"
              >
                <div
                  class="tag sc-ftvSup itDdzq"
                  data-emphasis="primary"
                  data-size="tiny"
                  data-variation="alternative"
                >
                  <p
                    class="paragraph "
                    data-size="tiny"
                    data-weight="bold"
                  >
                    Level 
                    beginner
                  </p>
                </div>
                <div
                  class="tag sc-ftvSup itDdzq"
                  data-emphasis="default"
                  data-size="tiny"
                  data-variation="default"
                >
                  <p
                    class="paragraph "
                    data-size="tiny"
                    data-weight="bold"
                  >
                    Impact 
                    high
                  </p>
                </div>
                <div
                  class="tag sc-ftvSup itDdzq"
                  data-emphasis="default"
                  data-size="tiny"
                  data-variation="alternative"
                >
                  <p
                    class="paragraph "
                    data-size="tiny"
                    data-weight="bold"
                  >
                    60
                     mins
                  </p>
                </div>
              </div>
            </div>
          </article>
          <article
            class="card false card--raised sc-eCYdqJ fmqJBb"
            data-testid="workout-card-4"
          >
            <img
              class="avatar sc-gKXOVf efHPNN"
              data-testid="thumbnail-Strength and Agility by Claire Donahoe"
              src="https://cgcdn.s3.amazonaws.com/nation/users/UID-D1489527062UID-58c86116311d9.png"
              style="width: 50px; height: 50px;"
            />
            <div
              class="sc-jSMfEi hEqNgO"
            >
              <p
                class="paragraph "
                data-size="default"
                data-weight="bold"
              >
                Strength and Agility by Claire Donahoe
              </p>
              <div
                class="sc-iBkjds cdystf"
              >
                <div
                  class="tag sc-ftvSup itDdzq"
                  data-emphasis="primary"
                  data-size="tiny"
                  data-variation="alternative"
                >
                  <p
                    class="paragraph "
                    data-size="tiny"
                    data-weight="bold"
                  >
                    Level 
                    beginner
                  </p>
                </div>
                <div
                  class="tag sc-ftvSup itDdzq"
                  data-emphasis="default"
                  data-size="tiny"
                  data-variation="default"
                >
                  <p
                    class="paragraph "
                    data-size="tiny"
                    data-weight="bold"
                  >
                    Impact 
                    low
                  </p>
                </div>
                <div
                  class="tag sc-ftvSup itDdzq"
                  data-emphasis="default"
                  data-size="tiny"
                  data-variation="alternative"
                >
                  <p
                    class="paragraph "
                    data-size="tiny"
                    data-weight="bold"
                  >
                    60
                     mins
                  </p>
                </div>
              </div>
            </div>
          </article>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dkzDqf uWKzM"
      data-testid="hero-section"
    >
      <h3
        class="heading sc-hKMtZM eZJtHV"
        data-font="united"
        data-variation="display"
      >
        Camp Gladiator is going to Mars...
      </h3>
      <h3
        class="heading sc-hKMtZM eZJtHV"
        data-font="united"
        data-variation="default"
      >
        Start now your virtual workouts!
      </h3>
    </div>
    <div
      class="sc-papXJ czGCkr"
    >
      <h3
        class="heading "
        data-font="gotham"
        data-variation="display"
      >
        Find the exercise that fits to you
      </h3>
      <form>
        <div
          class="fieldset fieldset--inline sc-crXcEl iRbwfs"
        >
          <div
            class="content sc-kDDrLX jUNfll"
          >
            <div
              class="outsideWrapper active"
              data-disabled="false"
              data-invalid="false"
              style="max-height: 75px;"
            >
              <div
                class="innerWrapper noRecords"
              >
                <p
                  class="paragraph placeholder"
                  data-size="small"
                  data-weight="book"
                >
                  Level
                </p>
                <div
                  class="wrapper paragraph"
                  data-size="small"
                  data-value=""
                >
                  <input
                    class="input"
                    value=""
                  />
                </div>
                <div
                  class="controls"
                />
              </div>
              <input
                hidden=""
                name="level"
                readonly=""
                value=""
              />
            </div>
          </div>
          <div
            class="content sc-kDDrLX jUNfll"
          >
            <div
              class="outsideWrapper active"
              data-disabled="false"
              data-invalid="false"
              style="max-height: 75px;"
            >
              <div
                class="innerWrapper noRecords"
              >
                <p
                  class="paragraph placeholder"
                  data-size="small"
                  data-weight="book"
                >
                  Impact
                </p>
                <div
                  class="wrapper paragraph"
                  data-size="small"
                  data-value=""
                >
                  <input
                    class="input"
                    value=""
                  />
                </div>
                <div
                  class="controls"
                />
              </div>
              <input
                hidden=""
                name="impact"
                readonly=""
                value=""
              />
            </div>
          </div>
          <button
            class="button sc-iqcoie fwqNIQ"
            data-children="true"
            data-emphasis="primary"
            data-floating="false"
            data-icon="false"
            data-size="default"
            data-variation="solid"
            type="submit"
          >
            <span
              class="label content"
              data-size="default"
            >
              Filter
            </span>
          </button>
        </div>
      </form>
      <div
        class="sc-evZas gzZmBv"
      >
        <article
          class="card false card--raised sc-eCYdqJ fmqJBb"
          data-testid="workout-card-3"
        >
          <img
            class="avatar sc-gKXOVf efHPNN"
            data-testid="thumbnail-Strength and Agility by Amber Autrey"
            src="https://cgcdn.s3.amazonaws.com/nation/users/UID-W1405610919UID-53c7eba7a992b.png"
            style="width: 50px; height: 50px;"
          />
          <div
            class="sc-jSMfEi hEqNgO"
          >
            <p
              class="paragraph "
              data-size="default"
              data-weight="bold"
            >
              Strength and Agility by Amber Autrey
            </p>
            <div
              class="sc-iBkjds cdystf"
            >
              <div
                class="tag sc-ftvSup itDdzq"
                data-emphasis="primary"
                data-size="tiny"
                data-variation="alternative"
              >
                <p
                  class="paragraph "
                  data-size="tiny"
                  data-weight="bold"
                >
                  Level 
                  beginner
                </p>
              </div>
              <div
                class="tag sc-ftvSup itDdzq"
                data-emphasis="default"
                data-size="tiny"
                data-variation="default"
              >
                <p
                  class="paragraph "
                  data-size="tiny"
                  data-weight="bold"
                >
                  Impact 
                  high
                </p>
              </div>
              <div
                class="tag sc-ftvSup itDdzq"
                data-emphasis="default"
                data-size="tiny"
                data-variation="alternative"
              >
                <p
                  class="paragraph "
                  data-size="tiny"
                  data-weight="bold"
                >
                  60
                   mins
                </p>
              </div>
            </div>
          </div>
        </article>
        <article
          class="card false card--raised sc-eCYdqJ fmqJBb"
          data-testid="workout-card-4"
        >
          <img
            class="avatar sc-gKXOVf efHPNN"
            data-testid="thumbnail-Strength and Agility by Claire Donahoe"
            src="https://cgcdn.s3.amazonaws.com/nation/users/UID-D1489527062UID-58c86116311d9.png"
            style="width: 50px; height: 50px;"
          />
          <div
            class="sc-jSMfEi hEqNgO"
          >
            <p
              class="paragraph "
              data-size="default"
              data-weight="bold"
            >
              Strength and Agility by Claire Donahoe
            </p>
            <div
              class="sc-iBkjds cdystf"
            >
              <div
                class="tag sc-ftvSup itDdzq"
                data-emphasis="primary"
                data-size="tiny"
                data-variation="alternative"
              >
                <p
                  class="paragraph "
                  data-size="tiny"
                  data-weight="bold"
                >
                  Level 
                  beginner
                </p>
              </div>
              <div
                class="tag sc-ftvSup itDdzq"
                data-emphasis="default"
                data-size="tiny"
                data-variation="default"
              >
                <p
                  class="paragraph "
                  data-size="tiny"
                  data-weight="bold"
                >
                  Impact 
                  low
                </p>
              </div>
              <div
                class="tag sc-ftvSup itDdzq"
                data-emphasis="default"
                data-size="tiny"
                data-variation="alternative"
              >
                <p
                  class="paragraph "
                  data-size="tiny"
                  data-weight="bold"
                >
                  60
                   mins
                </p>
              </div>
            </div>
          </div>
        </article>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
